### PR TITLE
packer: Add Docker composition

### PR DESCRIPTION
**Pull request by phlax**
Adds a docker composition that sets the versions of software used in build

In a follow up we can update READMEs/etc to point to the build here

### CODE REVIEW

File: docker/docker-compose.yaml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+  packer:
+    image: hashicorp/packer:latest
+    tty: true
+    volumes:
+    - ../:/workspace
+    - "${HOME}/.aws/credentials:/root/.aws/credentials"
+    working_dir: /workspace/ami-build
+    environment:
+      PACKER_PLUGIN_PATH: workspace/.packer.d/plugins
```
Comment (phlax) [inline docker/docker-compose.yaml]:
```suggestion
      PACKER_PLUGIN_PATH: /workspace/.packer.d/plugins
```

File: docker/docker-compose.yaml [hunk 2]
```diff
@@ -0,0 +1,42 @@
+services:
+
+  packer:
+    image: hashicorp/packer:latest
+    tty: true
```
Comment (phlax) [inline docker/docker-compose.yaml]: not sure if we need all these `tty`

**Comment by phlax** [comment]
@mum4k im also wondering about adding a build system more generally - im inclined to wrap the docker commands with `make` but then wondering about using `bazel`, potentially instead of docker if we can find recipes for it

**Comment by mum4k** [comment]
Thank you @phlax, to be able to review this, I would need a bit more context on what this is going to be used for. Is this a new feature, or are we considering using this to replace a portion of the existing setup? If the latter, what are we replacing?

**Comment by phlax** [comment]
> I would need a bit more context on what this is going to be used for. Is this a new feature, or are we considering using this to replace a portion of the existing setup?

to run the instructions in the README with updating, deterministic versions that simplifies running this - even/esp if the README specified versions, its unlikely im going to to go and try and get those versions for my host (even if i want to run them on my host)

**Comment by mum4k** [comment]
That is a very good idea, thank you for working on this!

**Review by mum4k — APPROVED** [review]

**Pull request merged**